zebra/kernel_netlink.c

```diff
@@ -1128,8 +1128,25 @@ static int nl_batch_read_resp(struct nl_batch *bth)
 	while (true) {
 		status = netlink_recv_msg(nl, msg, nl_batch_rx_buf,
 					  sizeof(nl_batch_rx_buf));
-		if (status == -1 || status == 0)
+		/*
+		 * status == -1 is a full on failure somewhere
+		 * since we don't know where the problem happened
+		 * we must mark all as failed
+		 *
+		 * Else we mark everything as worked
+		 *
+		 */
+		if (status == -1 || status == 0) {
+			while ((ctx = dplane_ctx_dequeue(&(bth->ctx_list)))
+			       != NULL) {
+				if (status == -1)
+					dplane_ctx_set_status(
+						ctx,
+						ZEBRA_DPLANE_REQUEST_FAILURE);
+				dplane_ctx_enqueue_tail(bth->ctx_out_q, ctx);
+			}
```

Comment:
I'd split `0` vs `-1`  in this case to their own if branches to avoid a second nested if and improve the readability a bit.

Reply:
I'm not going to create two branches here.  I don't think it creates readability at all.  Having 2 while loops and 2 dplane_ctx_enqueue_tail makes no sense to me

Reply:
my bad, I  thought we are not doing anything for the `0` case,  missed that the call to  `dplane_ctx_enqueue_tail` i not  under `status == -1`